In [5]:
import pandas as pd
pd.set_option('display.max_columns', None)

# Обзор данных

In [6]:
data = pd.read_csv('./features.csv', index_col='match_id')

In [7]:
print(data.head(10))

          start_time  lobby_type  r1_hero  r1_level  r1_xp  r1_gold  r1_lh  \
match_id                                                                     
0         1430198770           7       11         5   2098     1489     20   
1         1430220345           0       42         4   1188     1033      9   
2         1430227081           7       33         4   1319     1270     22   
3         1430263531           1       29         4   1779     1056     14   
4         1430282290           7       13         4   1431     1090      8   
5         1430284186           1       11         5   1961     1461     19   
8         1430293701           1        8         3    967     1136      7   
9         1430299335           7       35         5   2117     1252     16   
11        1430308974           1       17         5   1527      906     10   
12        1430316105           7       15         5   1651     1060     14   

          r1_kills  r1_deaths  r1_items  r2_hero  r2_level  r2_

In [8]:
dsc = data.describe()
dsc

start_time    lobby_type       r1_hero      r1_level         r1_xp  \
count  9.723000e+04  97230.000000  97230.000000  97230.000000  97230.000000   
mean   1.444232e+09      2.630999     51.517104      3.442672   1233.405801   
std    5.515393e+06      2.835761     32.564211      1.111741    566.588895   
min    1.430199e+09      0.000000      1.000000      0.000000      0.000000   
25%    1.440815e+09      1.000000     22.000000      3.000000    767.000000   
50%    1.446338e+09      1.000000     50.000000      3.000000   1175.000000   
75%    1.448829e+09      7.000000     75.000000      4.000000   1704.000000   
max    1.450313e+09      7.000000    112.000000      6.000000   3319.000000   

            r1_gold         r1_lh      r1_kills     r1_deaths      r1_items  \
count  97230.000000  97230.000000  97230.000000  97230.000000  97230.000000   
mean    1147.899702     11.231996      0.357009      0.362285      8.271315   
std      464.111662      9.041620      0.663889      0.626704      2.497575   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%      746.000000      2.000000      0.000000      0.000000      7.000000   
50%     1113.000000     11.000000      0.000000      0.000000      8.000000   
75%     1479.000000     19.000000      1.000000      1.000000     10.000000   
max     4332.000000     47.000000      8.000000      5.000000     34.000000   

            r2_hero      r2_level         r2_xp       r2_gold         r2_lh  \
count  97230.000000  97230.000000  97230.000000  97230.000000  97230.000000   
mean      52.183452      3.364661   1189.223676   1107.863993     10.471747   
std       32.674077      1.097536    555.363510    458.001007      8.972073   
min        1.000000      0.000000      0.000000      0.000000      0.000000   
25%       25.000000      3.000000    740.000000    708.000000      2.000000   
50%       50.000000      3.000000   1109.000000   1054.000000      9.000000   
75%       75.000000      4.000000   1641.000000   1434.000000     18.000000   
max      112.000000      6.000000   3234.000000   4136.000000     44.000000   

           r2_kills     r2_deaths      r2_items       r3_hero      r3_level  \
count  97230.000000  97230.000000  97230.000000  97230.000000  97230.000000   
mean       0.348709      0.363550      8.275584     52.710491      3.353924   
std        0.654060      0.624236      2.433864     32.560923      1.092126   
min        0.000000      0.000000      0.000000      1.000000      0.000000   
25%        0.000000      0.000000      7.000000     25.000000      3.000000   
50%        0.000000      0.000000      8.000000     51.000000      3.000000   
75%        1.000000      1.000000     10.000000     76.000000      4.000000   
max        8.000000      6.000000     33.000000    112.000000      7.000000   

              r3_xp       r3_gold         r3_lh      r3_kills     r3_deaths  \
count  97230.000000  97230.000000  97230.000000  97230.000000  97230.000000   
mean    1182.879965   1099.643742     10.333004      0.342723      0.357338   
std      554.899600    454.727127      8.950871      0.647774      0.618071   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%      736.000000    705.000000      2.000000      0.000000      0.000000   
50%     1101.000000   1040.000000      9.000000      0.000000      0.000000   
75%     1628.000000   1424.000000     18.000000      1.000000      1.000000   
max     3115.000000   3740.000000     51.000000      8.000000      6.000000   

           r3_items       r4_hero      r4_level         r4_xp       r4_gold  \
count  97230.000000  97230.000000  97230.000000  97230.000000  97230.000000   
mean       8.273527     52.648092      3.308896   1159.088481   1082.090240   
std        2.440139     32.670519      1.092502    550.937530    450.353291   
min        0.000000      1.000000      0.000000      0.000000      0.000000   
25%        7.000000     25.000000      3.000000    716

Описание признаков в таблице

    match_id: идентификатор матча в наборе данных
    start_time: время начала матча (unixtime)
    lobby_type: тип комнаты, в которой собираются игроки (расшифровка в dictionaries/lobbies.csv)
    Наборы признаков для каждого игрока (игроки команды Radiant — префикс rN, Dire — dN):
        r1_hero: герой игрока (расшифровка в dictionaries/heroes.csv)
        r1_level: максимальный достигнутый уровень героя (за первые 5 игровых минут)
        r1_xp: максимальный полученный опыт
        r1_gold: достигнутая ценность героя
        r1_lh: число убитых юнитов
        r1_kills: число убитых игроков
        r1_deaths: число смертей героя
        r1_items: число купленных предметов
    Признаки события "первая кровь" (first blood). Если событие "первая кровь" не успело произойти за первые 5 минут, то признаки принимают пропущенное значение
        first_blood_time: игровое время первой крови
        first_blood_team: команда, совершившая первую кровь (0 — Radiant, 1 — Dire)
        first_blood_player1: игрок, причастный к событию
        first_blood_player2: второй игрок, причастный к событию
    Признаки для каждой команды (префиксы radiant_ и dire_)
        radiant_bottle_time: время первого приобретения командой предмета "bottle"
        radiant_courier_time: время приобретения предмета "courier"
        radiant_flying_courier_time: время приобретения предмета "flying_courier"
        radiant_tpscroll_count: число предметов "tpscroll" за первые 5 минут
        radiant_boots_count: число предметов "boots"
        radiant_ward_observer_count: число предметов "ward_observer"
        radiant_ward_sentry_count: число предметов "ward_sentry"
        radiant_first_ward_time: время установки командой первого "наблюдателя", т.е. предмета, который позволяет видеть часть игрового поля
    Итог матча (данные поля отсутствуют в тестовой выборке, поскольку содержат информацию, выходящую за пределы первых 5 минут матча)
        duration: длительность
        radiant_win: 1, если победила команда Radiant, 0 — иначе
        Состояние башен и барраков к концу матча (см. описание полей набора данных)
            tower_status_radiant
            tower_status_dire
            barracks_status_radiant
            barracks_status_dire

In [14]:
rows = len(data)
counts = dsc.T['count']
counts_na = counts[counts < rows]
counts_na.sort_values().apply(lambda cnt: (rows - cnt)/rows)

first_blood_player2            0.452402
radiant_flying_courier_time    0.282619
dire_flying_courier_time       0.268415
first_blood_time               0.201100
first_blood_team               0.201100
first_blood_player1            0.201100
dire_bottle_time               0.166029
radiant_bottle_time            0.161380
radiant_first_ward_time        0.018883
dire_first_ward_time           0.018780
radiant_courier_time           0.007117
dire_courier_time              0.006953
Name: count, dtype: float64

# Обьяснение пропусков в данных

1) **first_blood_player2** - в 45% случаев firstblood совершаеться одним игроком <br>
2) **radiant_flying_courier_time, dire_flying_courier_time** в 28% и 26% соответственно команды не успевают покупать артефакт flying_courier в первые 5 минут <br>
3) **first_blood_time, first_blood_team, first_blood_player1** в 20% случаев firstblood не случаеться в первые 5 минут игры <br>
4) **dire_bottle_time, radiant_bottle_time** в ~16% случаев  команды не успевают покупать артефакт bottle в первые 5 минут игры <br> 
5) **radiant_first_ward_time, dire_first_ward_time** - в редких случаях ~1.8% команды не выставляют наблюдателя в первые 5 минут игры <br>
6) **radiant_courier_time, dire_courier_time** в < 1% случаев  команды не успевают покупать артефакт courier в первые 5 минут игры